Sources/Datadog/Core/Upload/HTTPHeadersProvider.swift

```diff
@@ -0,0 +1,96 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// Provides HTTP headers associated with SDK requests.
+internal struct HTTPHeadersProvider {
+    enum ContentType: String {
+        case applicationJSON = "application/json"
+        case textPlainUTF8 = "text/plain;charset=UTF-8"
+    }
+
+    struct HTTPHeader {
+        static let contentTypeHeaderField = "Content-Type"
+        static let userAgentHeaderField = "User-Agent"
+        static let ddAPIKeyHeaderField = "DD-API-KEY"
+        static let ddEVPOriginHeaderField = "DD-EVP-ORIGIN"
+        static let ddEVPOriginVersionHeaderField = "DD-EVP-ORIGIN-VERSION"
+        static let ddRequestIDHeaderField = "DD-REQUEST-ID"
+
+        enum Value {
+            /// If the header's value is constant.
+            case constant(_ value: String)
+            /// If the header's value is different each time.
+            case dynamic(_ value: () -> String)
+        }
+
+        let field: String
+        let value: Value
+
+        // MARK: - Standard Headers
+
+        /// Standard "Content-Type" header.
+        static func contentTypeHeader(contentType: ContentType) -> HTTPHeader {
+            return HTTPHeader(field: contentTypeHeaderField, value: .constant(contentType.rawValue))
+        }
+
+        /// Standard "User-Agent" header.
+        static func userAgentHeader(appName: String, appVersion: String, device: MobileDevice) -> HTTPHeader {
+            return HTTPHeader(
+                field: userAgentHeaderField,
+                value: .constant("\(appName)/\(appVersion) CFNetwork (\(device.model); \(device.osName)/\(device.osVersion))")
+            )
+        }
+
+        // MARK: - Datadog Headers
+
+        /// Datadog request authentication header.
+        static func ddAPIKeyHeader(clientToken: String) -> HTTPHeader {
```

Comment:
this might be confusing for the outsiders/newcomers as we have `client token`s and [`API key`s](https://app.datadoghq.com/organization-settings/api-keys)

what do you think of renaming this function to `ddClientTokenHeader`? (of course we will keep `"DD-API-KEY"` as value)

Reply:
Here I used consistent convention for 3 new headers:
```swift
static func ddAPIKeyHeader(clientToken: String) -> HTTPHeader
static func ddEVPOriginHeader(source: String) -> HTTPHeader
static func ddEVPOriginVersionHeader() -> HTTPHeader
```
to make it clearly visible that we follow the spec for all features:
```swift
// in LoggingFeature, TracingFeature, RUMFeature and InternalMonitoringFeature:
requestBuilder: RequestBuilder(
   // ...
   headers: [
      // ...
      .ddAPIKeyHeader(clientToken: configuration.clientToken),
      .ddEVPOriginHeader(source: configuration.common.source),
      .ddEVPOriginVersionHeader(),
   ]
)
```
The spec says we must send `DD-API-KEY`, `DD-EVP-ORIGIN` and `DD-EVP-ORIGIN-VERSION` headers - and I shaped the above code to tell exactly that story.

IMO if we reverse it by naming the function `ddClientTokenHeader()`, the `DD-API-KEY` value used internally would be equally puzzling to the reader.